code/game/objects/effects/effect_system/effect_system.dm

```diff
@@ -85,28 +85,40 @@ steam.start() -- spawns the effect
 /obj/effect/particle_effect/sparks
 	name = "sparks"
 	icon_state = "sparks"
-	var/amount = 6
 	anchored = TRUE
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	var/amount = 6
+	var/sound_to_play = "sparks"
 
 /obj/effect/particle_effect/sparks/New()
 	..()
-	playsound(src.loc, "sparks", 25, 1)
+	if(sound_to_play)
+		playsound(src.loc, sound_to_play, 25, 1)
 // var/turf/T = src.loc
 // if (istype(T, /turf))
 // T.hotspot_expose(1000,100)
 	spawn (100)
 		qdel(src)
 
+/obj/effect/particle_effect/sparks/armor_shards
+	name = "armor shards"
+	icon_state = "armor_shards"
+	sound_to_play = null
+	amount = 4
 
 /datum/effect_system/spark_spread
 	var/total_sparks = 0 // To stop it being spammed and lagging!
-
-/datum/effect_system/spark_spread/set_up(n = 3, c = 0, loca)
-	if(n > 10)
-		n = 10
-	number = n
-	cardinals = c
+	/// Amount of sleep ticks before the sparks start moving around.
+	var/spark_sleep_delay = 5
+	/// How long sparks stay on the ground after creation.
+	var/spark_duration = 2 SECONDS
+	var/obj/effect/particle_effect/sparks/sparks_type = /obj/effect/particle_effect/sparks
```

Comment:
code doc + why is this a casted var when it's a type that never utilizes initial() anywhere